src/SigSpec.AspNetCore/SigSpecDocumentRegistration.cs

```diff
@@ -0,0 +1,33 @@
+﻿using SigSpec.AspNetCore.Middlewares;
+using SigSpec.Core;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>A SigSpec document generator registration.</summary>
+    internal class SigSpecDocumentRegistration
+    {
+        private SigSpecDocumentGeneratorSettings _settings;
+
+        /// <summary>Initializes a new instance of the <see cref="SigSpecDocumentRegistration"/> class.</summary>
+        /// <param name="settings">The document settings.</param>
+        public SigSpecDocumentRegistration(SigSpecDocumentGeneratorSettings settings)
+        {
+            _settings = settings;
+        }
+
+        /// <summary>Gets the document name.</summary>
+        public string DocumentName => _settings.DocumentName;
+
+        /// <summary>
+        /// Generates the document.
+        /// </summary>
+        /// <returns>The JSON.</returns>
+        public async Task<string> GenerateJsonAsync()
+        {
+            var generator = new SigSpecGenerator(_settings);
+            var document = await generator.GenerateForHubsAsync(_settings.Hubs, _settings.Template);
```

Comment:
I noticed that when using the _setting.Template the definitions gets larger everytime
![image](https://user-images.githubusercontent.com/6969872/82690512-465e6a00-9c54-11ea-9e0e-035d8c82b7e2.png)
 

I think that got introduced recently, not sure if you have any ideas about that?

Reply:
Oh yes, it will just reuse that and extend it more and more which is bad.. we need to create a copy of the template (tojson/fromjson) before we generate stuff into it